### PR TITLE
Refresh plugin for october 2023

### PR DESCRIPTION
**Pull request by jonesbusy**
Hi,

This is a refresh of the plugin

More info about the changes: https://www.jenkins.io/doc/developer/tutorial-improve/

It's important to depends on recent/support core version to avoid implicit dependencies : https://plugins.jenkins.io/mstest/dependencies/

Build might fail because need to be restarted by a maintainer or admin.

### Testing done

`mvn clean install`

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by jonesbusy** [comment]
Can an admin restart the build to apply Jenkinsfile changes ? Thanks!

**Comment by jonesbusy** [comment]
Hi @pavlovic-ivan

Can you take a look at this ?

Due to Jenkinsfile changes an admin or maintainer should restart the build to take Jenkinsfile changes.

I think you should be able to to it from the action tab : https://github.com/jenkinsci/mstest-plugin/pull/24/checks?check_run_id=17949982213

Thanks!

File: pom.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <version>${revision}${changelist}</version>
     <url>http://wiki.jenkins-ci.org/display/JENKINS/MSTest+Plugin</url>
     <scm>
-        <connection>scm:git:git@github.com:jenkinsci/mstest-plugin.git</connection>
+        <connection>scm:git:https@github.com:jenkinsci/mstest-plugin.git</connection>
```
Comment (MarkEWaite) [inline pom.xml:22]: Need to use a valid URL.
```suggestion
        <connection>scm:git:https://github.com:jenkinsci/mstest-plugin.git</connection>
```

**Review by MarkEWaite — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jonesbusy) [reply pom.xml]: Good catch thanks!

**Comment by pavlovic-ivan** [comment]
Hi @jonesbusy . I am taking a look now. Sorry for the delay

**Comment by jonesbusy** [comment]
Thanks you @pavlovic-ivan 

Mark restarted the build so it's now building on JDK17 and JDK20 correctly

I would appreciate a lot a release with those in order to start using the plugin

**Comment by pavlovic-ivan** [comment]
@jonesbusy release is done. new version is `1.0.4`

**Pull request merged**